default-fields/0000-default-fields.md

```diff
@@ -6,31 +6,201 @@
 [motivation]: #motivation
 
-Why are we doing this? What use cases does it support? What is the expected outcome?
+Today, Rust allows you to create an instance of a struct using a literal syntax.
+This requires all fields in the struct be assigned a value, and can't be used
+with private fields.
+It can also be inconvenient for large structs that usually receive the same values.
+
+With the `..` syntax, values for missing fields can be taken from another struct.
+This is convenient for using `Default` to override just a few values on initialisation:
+
+```rust
+#[derive(Default)]
+struct Foo {
+    a: &'static str,
+    b: bool,
+    c: i32,
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+    ..Default::default()
+}
+```
+
+However, this still requires an already initialised struct after the `..`.
+
+To work around these shortcomings, users can create constructor functions or more elaborate builders.
+The problem with a constructor is that you need one for each combination of fields a caller can supply.
+Builders enable more advanced initialisation, but need additional boilerplate.
 
 # Detailed design
 [design]: #detailed-design
 
-This is the bulk of the RFC. Explain the design in enough detail for somebody familiar
-with the language to understand, and for somebody familiar with the compiler to implement.
-This should get into specifics and corner-cases, and include examples of how the feature is used.
+With field defaults a caller can initialise a struct with default values without needing builders
+or a constructor function:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32,
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+}
+```
+
+Field defaults are sugar for the 'real' initialiser, where values for missing fields are added with the supplied default expression.
+
+Supplied field values take precedence over field defaults:
+
+```rust
+// `b` is `false`, even though the field default is `true`
+let foo = Foo {
+    a: "Hello",
+    b: false,
+    c: 42,
+};
+```
+
+Supplied field values in functional updates take precedence over field defaults:
+
+```rust
+// `b` is `false`, even though the field default is `true`
+let foo = Foo {
+    a: "Hello",
+    c: 0,
+    ..Foo {
+        b: false
+    }
+};
+```
+
+The field default should only be used when the caller doesn't supply a value,
+to avoid unnecessarily assigning values.
+
+When deriving `Default`, field defaults are used instead of the type default.
+
+```rust
+#[derive(Default)]
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32,
+}
+
+// `b` is `true`, even though `bool::default()` is `false`
+let foo = Foo::default();
+```
+
+## Allowable Values
+
+Field defaults use the same syntax as `const`s.
+So the type must be supplied, and the value must be a compile-time expression, or a call to `Default::default()`.
+Default field values are expected to be cheap to produce and have basic support for collection types like `Vec`.
```

Comment:
Note this is a bit different now that default is in alternatives. I've added a point that the syntax and functionality are modeled after `const`. Maybe it could be clearer though, since the syntax is in the grammar section, and the functionality is in interpretation.